python/ray/data/_internal/pandas_block.py

```diff
@@ -134,7 +134,15 @@ def _empty_table() -> "pandas.DataFrame":
 class PandasBlockAccessor(TableBlockAccessor):
     ROW_TYPE = PandasRow
 
-    def __init__(self, table: "pandas.DataFrame"):
+    def __init__(self, table: Union["pandas.DataFrame", "pyarrow.Table"]):
+        from pyarrow import Table
```

Comment:
Will lazily import `pyarrow.Table` after we merge https://github.com/ray-project/ray/pull/39817 to master